[CI/Build] add EP dependencies to docker

zhewenl opened the pull request:
## Purpose
This PR adds EP dependencies (EP kernels(pplx-kernels and DeepEP), NixL and DeepGEMM)to docker, so that we can test H200 in buildkite like https://github.com/vllm-project/vllm/pull/20733

## Test Plan
manually trigger CI: https://buildkite.com/vllm/ci/builds/25528#01985e90-88cd-4d3f-8bcc-a362a5c1a93e

## Test Result
<img width="1720" height="411" alt="Screenshot 2025-07-30 at 11 04 29 PM" src="https://github.com/user-attachments/assets/9178cff7-557b-4dec-9713-4d492f5af9e2" />

tested in this build: https://buildkite.com/vllm/ci/builds/27389
(tested using draft PR: https://github.com/vllm-project/vllm/pull/23080)

zhewenl commented on docker/Dockerfile:
```diff
@@ -429,6 +429,17 @@ RUN --mount=type=cache,target=/root/.cache/uv \
+# Install EP kernels(pplx-kernels and DeepEP), NixL and DeepGEMM
+COPY tools/ep_kernels/install_python_libraries.sh install_python_libraries.sh
+COPY tools/install_nixl.sh install_nixl.sh
+COPY tools/install_deepgemm.sh install_deepgemm.sh
```
perhaps we can just add deepgemm installation to install_python_libraries.sh

simon-mo commented on docker/Dockerfile:
```diff
@@ -429,6 +429,17 @@ RUN --mount=type=cache,target=/root/.cache/uv \
+    && bash install_deepgemm.sh \
+    && bash install_nixl.sh --force \
+    # remove libraries after installation
+    && rm -rf ep_kernels_workspace deepgemm_workspace nixl_installer
```
I thought we need these workspace because the required `so` files and others are built in there.

zhewenl committed:
support ep/dp

zhewenl committed:
update deps

zhewenl committed:
update deps

simon-mo reviewed (commented):
Please add corresponding tests showing it work, do we need to the workspace directory after these, if not, let's clean it up.

simon-mo committed:
Merge branch 'main' into add-ep-dp-kernels

zhewenl committed:
add qwen ep test

zhewenl commented:
updated PR and added a dummy test in current CI to validate DeepEP/DeepGemm: https://buildkite.com/vllm/ci/builds/27781#0198c89a-53d4-451c-b539-ef6ceef6ece0

Will iterate on it further with more proper tests

simon-mo commented:
Nice. I can see https://buildkite.com/vllm/ci/builds/27781/steps/canvas?sid=0198c89a-52ad-4904-9b9e-5fad172c2bb0
